src/main/java/servlets/UploadReceiptServlet.java

```diff
@@ -273,6 +280,13 @@ private String getBaseUrl(HttpServletRequest request) {
     return baseUrl;
   }
 
+  /**
+   * Gets the list of user-assigned categories from the request.
+   */
+  private Set<String> getCategories(HttpServletRequest request) {
```

Comment:
Use ImmutableSet.

As Effective Java Item 17 explains, "immutable classes are easier to design, implement, and use than mutable classes. They are less prone to error and are more secure."